src/createEventEmitterProxy.test.ts

```diff
@@ -160,4 +203,93 @@ describe('createEventEmitterProxy', () => {
 
     expect(proxy.bar()).toStrictEqual([true, 42]);
   });
+
+  it('only migrates events that were added via the proxy, not all the events', () => {
+    const original = new EventEmitter();
+    const next = new EventEmitter();
+
+    let moveCount = 0;
+    let dontMoveCount = 0;
+    original.on('shouldNotMove', () => (dontMoveCount += 1));
+
+    const proxy = createEventEmitterProxy(original);
+    expect(proxy.eventNames()[0]).toBe('shouldNotMove');
+    proxy.on('shouldMove', () => (moveCount += 1));
+    proxy.setTarget(next);
+    next.emit('shouldMove');
+    expect(moveCount).toBe(1);
+    next.emit('shouldNotMove');
+    expect(dontMoveCount).toBe(0);
+  });
+
+  it('does not migrate events that have been removed via "off"', () => {
+    const original = new EventEmitter();
+    const next = new EventEmitter();
+
+    const proxy = createEventEmitterProxy(original);
+    let count = 0;
+    const inc = () => (count += 1);
+    proxy.on('foo', inc);
+    original.emit('foo');
+    expect(count).toBe(1);
+    proxy.off('foo', inc);
+    original.emit('foo');
+    expect(count).toBe(1);
+    proxy.setTarget(next);
+    next.emit('foo');
+    expect(count).toBe(1);
+  });
+
+  it('does not migrate events that have been removed via "removeListener"', () => {
+    const original = new EventEmitter();
+    const next = new EventEmitter();
+
+    const proxy = createEventEmitterProxy(original);
+    let count = 0;
+    const inc = () => (count += 1);
+    proxy.on('foo', inc);
+    original.emit('foo');
+    expect(count).toBe(1);
+    proxy.removeListener('foo', inc);
+    original.emit('foo');
+    expect(count).toBe(1);
+    proxy.setTarget(next);
+    next.emit('foo');
+    expect(count).toBe(1);
+  });
+
+  it('can set properties on the proxied event emitter', () => {
+    const original = new EventEmitter();
+    const proxy = createEventEmitterProxy(original);
+    (proxy as any).foo = 123;
+    expect((original as any).foo).toBe(123);
+  });
+
+  it('can get values that are non-functions', () => {
+    const original = new EventEmitter();
+    const proxy = createEventEmitterProxy(original);
+    expect((proxy as any)._eventsCount).toBe(0);
+  });
+
+  it('can set a custom "setTarget" method', () => {
+    const original = new EventEmitter();
+    const proxy = createEventEmitterProxy(original);
+    const mockSetTarget = jest.fn();
+    proxy.setTarget = mockSetTarget;
+
+    const next = new EventEmitter();
+    proxy.setTarget(next);
+    expect(mockSetTarget).toHaveBeenCalledWith(next);
+  });
+
+  it('can handle the context being the proxy itself', () => {
```

Comment:
Could you clarify what the goal of this test is?

It looks like we're calling `on` and explicitly setting the `thisArg` to the context for the test itself. But the expectation is just checking what `this` is within the listener, which seems unrelated to the `thisArg` used for the call

Reply:
It definitely seems worth checking that `this` is set correctly for unbound listeners, but perhaps we can make this test more straightforward by removing step where `call` is used to change `thisArg`. But perhaps I'm missing something.

Reply:
This is just to get coverage on the line `return value.apply(this === receiver ? target : this, args);`
Im not sure why we needed to handle this, but someone added it and didnt test it, and it seemed easy enough to at least add a test that hits that line. Feel free to make a new issue about adding a better description / better test case for this, and maybe someone with more context around this ternary can write a better test for it.

Reply:
Ah darn, sorry I was getting mixed up, I was thinking of how the event listeners are always bound to the event emitter. But this isn't the listener, this is the call to add one. This all makes sense, nevermind.

Reply:
ahh gotcha. Yeah... NGL its pretty confusing up in here. Hard to keep track of what is what. Might be good to heavily JSdoc this puppy in a followup.